lib/internal/Magento/Framework/Stdlib/DateTime/Timezone.php

```diff
@@ -318,4 +328,19 @@ public function convertConfigTimeToUtc($date, $format = 'Y-m-d H:i:s')
 
         return $date->format($format);
     }
+
+    /**
+     * Retrieve date with time
+     *
+     * @param Boolean $includeTime
+     * @param String $date
+     * @return  String
+     */
+    private function getDateWithTime($includeTime, $date)
+    {
+        if ($includeTime && !preg_match('/(.*).(\d{2}):(\d{2})/', $date)) {
```

Comment:
Please simplify regular expression as suggested in https://github.com/magento/magento2/pull/11306#discussion_r153326931

Reply:
@orlangur I have tried do it how you say, but it doesn't work.

Reply:
You mean the `(.*)` part is essential? At least unneeded brackets can be removed.

Reply:
Ah ok, I'm going to check again because I thought you were referring to something else.

Reply:
I forgot delimiter there, sorry
```
!preg_match('\d{2}:\d{2}', $date)
```
so 
```
!preg_match('/\d{2}:\d{2}/', $date)
```
should work I believe

Reply:
I think that the is best this
```!preg_match('/\d{1}:\d{2}/', $date)``` 
In this way we check 1:00am also.

Reply:
Agree 👍